Writer: Loggo emoji writer

SimonRichardson opened the pull request:
The following introduces an emoji log writer. For fun and profit.


![Screenshot from 2021-07-02 13-01-15](https://user-images.githubusercontent.com/2562584/124272863-398a9600-db37-11eb-8a9e-9a97940da9b2.png)

SimonRichardson committed:
Writer: Loggo emoji writer

The following introduces a emoji log writer. For fun and profit.

erik78se commented:
This is fantastic!

jameinel reviewed (changes requested):
needs moar tests

erik78se commented:
How can I test this?I have a charm that already logs with this internally and I would love to be able to use this also in conjunction with stuff like "juju debug-log". The visibility for specific log events helps alot to discover when stuff goes sideways or even just info.

erik78se commented:
I think that "juju debug-log" will have a hard time here perhaps, thats one thing I would like to test etc.

SimonRichardson commented:
@erik78se we'll just ensure that it gets written, won't take me long... will get it done Monday :dancers:

SimonRichardson committed:
Emoji: Add tests for the emoji writer

The following ensures that we actually write out a emoji for the correct
level.

erik78se commented:
When can I use this ?

SimonRichardson commented:
I need to come back around and merge this into Juju.

erik78se commented:
Cool, how do I add these icons myself in my logs if I'd like to lets say
add a charm-custom emoji?

logger.info( ????? )

On Tue, Aug 10, 2021 at 12:25 PM Simon Richardson ***@***.***>
wrote:

> I need to come back around and merge this into Juju.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/juju/loggo/pull/41#issuecomment-895913869>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAGPVK7EN7LZQ3ESKMY3D7LT4D5CNANCNFSM47WTIYAA>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&utm_campaign=notification-email>
> .
>

jameinel commented:
your charm can add whatever characters it wants to its log lines. Something
like:

log.warning("\uemoji my message")

for whatever the codes are for the emojis you want.

On Tue, Aug 10, 2021, 06:49 Erik Lönroth ***@***.***> wrote:

> Cool, how do I add these icons myself in my logs if I'd like to lets say
> add a charm-custom emoji?
>
> logger.info( ????? )
>
> On Tue, Aug 10, 2021 at 12:25 PM Simon Richardson ***@***.***>
> wrote:
>
> > I need to come back around and merge this into Juju.
> >
> > —
> > You are receiving this because you were mentioned.
> > Reply to this email directly, view it on GitHub
> > <https://github.com/juju/loggo/pull/41#issuecomment-895913869>, or
> > unsubscribe
> > <
> https://github.com/notifications/unsubscribe-auth/AAGPVK7EN7LZQ3ESKMY3D7LT4D5CNANCNFSM47WTIYAA
> >
> > .
> > Triage notifications on the go with GitHub Mobile for iOS
> > <
> https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675
> >
> > or Android
> > <
> https://play.google.com/store/apps/details?id=com.github.android&utm_campaign=notification-email
> >
> > .
> >
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/juju/loggo/pull/41#issuecomment-895927429>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AABRQ7ODXDGJTQYVDVRHOHDT4D72PANCNFSM47WTIYAA>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&utm_campaign=notification-email>
> .
>

erik78se commented:
Right, I'll try it out =)

On Tue, Aug 10, 2021 at 1:23 PM John Arbash Meinel ***@***.***>
wrote:

> your charm can add whatever characters it wants to its log lines. Something
> like:
>
> log.warning("\uemoji my message")
>
> for whatever the codes are for the emojis you want.
>
> On Tue, Aug 10, 2021, 06:49 Erik Lönroth ***@***.***> wrote:
>
> > Cool, how do I add these icons myself in my logs if I'd like to lets say
> > add a charm-custom emoji?
> >
> > logger.info( ????? )
> >
> > On Tue, Aug 10, 2021 at 12:25 PM Simon Richardson ***@***.***>
> > wrote:
> >
> > > I need to come back around and merge this into Juju.
> > >
> > > —
> > > You are receiving this because you were mentioned.
> > > Reply to this email directly, view it on GitHub
> > > <https://github.com/juju/loggo/pull/41#issuecomment-895913869>, or
> > > unsubscribe
> > > <
> >
> https://github.com/notifications/unsubscribe-auth/AAGPVK7EN7LZQ3ESKMY3D7LT4D5CNANCNFSM47WTIYAA
> > >
> > > .
> > > Triage notifications on the go with GitHub Mobile for iOS
> > > <
> >
> https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675
> > >
> > > or Android
> > > <
> >
> https://play.google.com/store/apps/details?id=com.github.android&utm_campaign=notification-email
> > >
> > > .
> > >
> >
> > —
> > You are receiving this because you commented.
> > Reply to this email directly, view it on GitHub
> > <https://github.com/juju/loggo/pull/41#issuecomment-895927429>, or
> > unsubscribe
> > <
> https://github.com/notifications/unsubscribe-auth/AABRQ7ODXDGJTQYVDVRHOHDT4D72PANCNFSM47WTIYAA
> >
> > .
> > Triage notifications on the go with GitHub Mobile for iOS
> > <
> https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675
> >
> > or Android
> > <
> https://play.google.com/store/apps/details?id=com.github.android&utm_campaign=notification-email
> >
> > .
> >
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/juju/loggo/pull/41#issuecomment-895948649>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAGPVK7POXNPTSBC6JIHSPDT4ED45ANCNFSM47WTIYAA>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&utm_campaign=notification-email>
> .
>
